xdgdesktopfile.cpp

```diff
@@ -770,6 +771,33 @@ QString const XdgDesktopFile::iconName() const
 }
 
 
+QStringList XdgDesktopFile::mimeTypes() const
+{
+    if (contains(mimeTypeKey))
+    {
+        QStringList s = value(mimeTypeKey).toString().split(QLatin1Char(';'));
+        if (s.isEmpty())
+        {
+            return s;
+        }
+        else
+        {
+            // Remove empty strings
+            QMutableListIterator<QString> it(s);
```

Comment:
I was always curious about this Qt iterating style.... Especialy here, is there any benefit over the stl-like manipulation?
E.g.:
- pre C++11
```diff
diff --git a/xdgdesktopfile.cpp b/xdgdesktopfile.cpp
index b27070c..f379aaf 100644
--- a/xdgdesktopfile.cpp
+++ b/xdgdesktopfile.cpp
@@ -33,6 +33,7 @@
 
 #include <stdlib.h>
 #include <unistd.h>
+#include <functional>
 
 #include <QDebug>
 #include <QDBusInterface>
@@ -783,13 +784,7 @@ QStringList XdgDesktopFile::mimeTypes() const
         else
         {
             // Remove empty strings
-            QMutableListIterator<QString> it(s);
-            while (it.hasNext())
-            {
-                const QString dir = it.next();
-                if (dir.isEmpty())
-                    it.remove();
-            }
+            s.erase(std::remove_if(s.begin(), s.end(), std::mem_fun_ref(&QString::isEmpty)), s.end());
         }
         return s;
     }
```
- post C++11
```diff
diff --git a/xdgdesktopfile.cpp b/xdgdesktopfile.cpp
index b27070c..9d1e759 100644
--- a/xdgdesktopfile.cpp
+++ b/xdgdesktopfile.cpp
@@ -33,6 +33,7 @@
 
 #include <stdlib.h>
 #include <unistd.h>
+#include <functional>
 
 #include <QDebug>
 #include <QDBusInterface>
@@ -783,13 +784,7 @@ QStringList XdgDesktopFile::mimeTypes() const
         else
         {
             // Remove empty strings
-            QMutableListIterator<QString> it(s);
-            while (it.hasNext())
-            {
-                const QString dir = it.next();
-                if (dir.isEmpty())
-                    it.remove();
-            }
+            s.erase(std::remove_if(std::begin(s), std::end(s), std::bind(&QString::isEmpty, std::placeholders::_1)), std::end(s));
         }
         return s;
     }
```